### PR TITLE
Move to prerelease refs in netstandard spec for CA.Common nuspec

### PR DESCRIPTION
**Pull request by agocke**

### CODE REVIEW

**Comment by agocke** [comment]
Still testing this locally, please don't merge yet.

**Comment by tmat** [comment]
Urm, this is kind of a lie, right? I mean we do not depend on immutable 1.2.0 yet. That's what my PR does: https://github.com/dotnet/roslyn/pull/10307

**Comment by agocke** [comment]
@tmat It doesn't matter. `FileVersionInfo` depoends on 1.2.0-rc2-23931, so we depend on it indirectly.

**Comment by tmat** [comment]
We should keep the $$ values as they are.

**Comment by tmat** [comment]
I see. OK. I guess we can use the higher version on Core.

**Comment by tmat** [comment]
:+1:

File: src/NuGet/Microsoft.CodeAnalysis.Common.nuspec [hunk 1]
```diff
@@ -19,13 +19,12 @@ Supported Platforms:
         </group>
 
         <group targetFramework="netstandard1.3">
-            <dependency id="System.Collections.Immutable" version="$systemCollectionsImmutableVersion$" />
-            <dependency id="System.Reflection.Metadata" version="$systemReflectionMetadataVersion$" />
             <dependency id="Microsoft.CodeAnalysis.Analyzers" version="$codeAnalysisAnalyzersVersion$" />
```
Comment (tmat) [inline src/NuGet/Microsoft.CodeAnalysis.Common.nuspec]: Why do we have a dependency on analyzers?

**Comment by agocke** [comment]
Managed to reproduce the downgrade warning without this change and it's fixed with this change, so this should be good to merge.

**Comment by agocke** [comment]
(Ignoring the unit tests, since this is a NuGet-only change).

File: src/NuGet/Microsoft.CodeAnalysis.Common.nuspec (hunk 1 above)
Comment (weshaggard) [inline src/NuGet/Microsoft.CodeAnalysis.Common.nuspec]: I would also like to understand this dependency.

**Comment by weshaggard** [comment]
LGTM

**Comment by chriswelles** [comment]
Using Microsoft.Net.Compilers 1.3.0 forces the use of the older 1.2.0 System.Reflection.Metadata.  This package is trying to force it up upgrade to 1.3.0, which just doesn't work.  Things are broken.

**Comment by agocke** [comment]
@chriswelles I'm not sure what problem you're experiencing -- when I try to build an analyzer against Microsoft.CodeAnalysis.CSharp v1.3.0 I get references to S.R.M 1.2.0 and System.Collections.Immutable 1.1.37. Those references look right to me, what's wrong with them for you?

Also, I'd recommend building against older versions if you're building an analyzer. If you build against version 1.3.* of the CodeAnalysis assemblies you'll only work on VS 2015 Update 3 and later, but if you build against version 1.0.*, you'll run on all versions of VS 2015.

**Comment by chriswelles** [comment]
The 1.3.0 CodeAnalysis.Common nuget package references the nuget for "S.R.M" 1.3.0 and "S.C.I" 1.2.0.  The Microsoft.Net.Compilers forces it's own version of the assemblies down, despite the nuget, HintPath, etc.  The nuget for the newer versions sometimes adds assembly redirects breaking things.  

When I reference CodeAnalysis on it's own, I get S.R.M. 1.3.0.  When I then Microsoft.Net.Compilers on top of that, I get S.R.M 1.2.0 (although the S.R.M. 1.3.0 nuget is still referenced).  Two issues, which don't seem to be too much of a problem on their own, are interacting and causing actual issues.

**Comment by agocke** [comment]
@chriswelles I'm not sure how that could happen. Microsoft.Net.Compilers 1.3.0 has no dependencies: https://www.nuget.org/packages/Microsoft.Net.Compilers/1.3.0

**Comment by chriswelles** [comment]
That's true from a nuget perspective.  Somehow, with a .csproj project, Microsoft.Net.Compilers is forcing it's own version of the assemblies down.  I don't understand why, but it does that, despite any HintPath in the csproj, nuget files referenced, etc.  See issue 12255 where a number of us seem to have been bitten by this.

With a newer xproj project, I don't seem to be hitting this issue.  It's pulling in the file version referenced by the nuget.

**Pull request merged**